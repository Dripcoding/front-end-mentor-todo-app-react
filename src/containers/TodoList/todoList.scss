@import "../../mixins";
@import "../../palette.scss";

.todoList__container {
  @include flex(column, flex-start, center);

  background-color: inherit;
  height: 439px;
  width: 540px;

}