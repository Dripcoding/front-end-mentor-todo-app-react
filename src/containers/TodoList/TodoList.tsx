import React, { useContext } from "react";
import { TodoContext } from "../../context/TodoContext";
import TodoListItem from "../../components/TodoListItem";

import "./todoList.scss";

const STYLE_BASE: string = "todoList__";

const TodoList = (): JSX.Element => {
  const { state, dispatch } = useContext(TodoContext);
  const { todos } = state;

  return (
    <div className={`${STYLE_BASE}container`} data-testid={"TODO_LIST_CONTAINER"}>
      {todos.map(todo => (
        <TodoListItem todo={todo} />
      ))}
    </div>
  );
};

export default TodoList;