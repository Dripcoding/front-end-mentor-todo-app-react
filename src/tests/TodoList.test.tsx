import { render } from "@testing-library/react";
import TodoList from "../containers/TodoList/index";

describe("TodoList", () => {
  it("should match snapshot", () => {
    const { container } = render(<TodoList />);
    expect(container).toMatchSnapshot();
  });

  it("should render correctly", () => {
    const { getByTestId } = render(<TodoList />);

    const container = getByTestId("TODO_LIST_CONTAINER");
    expect(container).toBeInTheDocument();
    expect(container).toBeVisible();
  });
});