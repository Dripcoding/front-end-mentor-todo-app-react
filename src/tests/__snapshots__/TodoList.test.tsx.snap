// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should match snapshot 1`] = `
<div>
  <div
    class="todoList__container"
    data-testid="TODO_LIST_CONTAINER"
  />
</div>
`;
